cpp/src/neighbors/ivf_flat/ivf_flat_search.cuh

```diff
@@ -402,15 +403,25 @@ void search(raft::resources const& handle,
             const index<T, IdxT>& idx,
             raft::device_matrix_view<const T, IdxT, raft::row_major> queries,
             raft::device_matrix_view<IdxT, IdxT, raft::row_major> neighbors,
-            raft::device_matrix_view<float, IdxT, raft::row_major> distances)
+            raft::device_matrix_view<float, IdxT, raft::row_major> distances,
+            cuvs::neighbors::filtering::base_filter* sample_filter_ptr)
 {
-  search_with_filtering(handle,
-                        params,
-                        idx,
-                        queries,
-                        neighbors,
-                        distances,
-                        cuvs::neighbors::filtering::none_ivf_sample_filter());
+  if (sample_filter_ptr == nullptr ||
```

Comment:
Is the ptr absolutely needed? Can't we pass by reference and then dereference internally if we really need to have a pointer available? Pointers are messy and error prone. When working w/ C++ objects, I'd much rather just pass by reference everywhere.